user/contactmethod/store.go

```diff
@@ -466,3 +477,39 @@ func (db *DB) FindAll(ctx context.Context, userID string) ([]ContactMethod, erro
 
 	return scanAll(rows)
 }
+
+func (db *DB) FindLastStatus(ctx context.Context, cmID string) (*n.MessageStatus, error) {
+	err := validate.UUID("Contact Method ID", cmID)
+	if err != nil {
+		return nil, err
+	}
+
+	err = permission.LimitCheckAny(ctx, permission.User)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := db.db.BeginTx(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	var s n.MessageStatus
+	var lastStatus string
+	var hasNextRetry bool
+	row := wrapTx(ctx, tx, db.findLastStatus).QueryRowContext(ctx, cmID)
+	err = row.Scan(&lastStatus, &s.Details, &hasNextRetry)
+	if err != nil {
+		return nil, err
+	}
+
+	state := n.GetMessageStatusState(lastStatus, hasNextRetry)
+	if state == -1 {
+		return nil, fmt.Errorf("unknown last_status %s", lastStatus)
+	}
+
+	s.State = state
+
+	return &s, nil
```

Comment:
While we can do away with the transaction -- you should always commit within the same scope you call `BeginTx` (other than special circumstances).

If you do use a tx for a read operation, the results are only guaranteed valid (for whatever isolation level is chosen) if the commit succeeds.